rust/rpmostree-client/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+//! APIs for interacting with rpm-ostree client side.
+
+use anyhow::Context;
+use serde_derive::Deserialize;
+use std::process::Command;
+
+/// Our generic catchall fatal error, expected to be converted
+/// to a string to output to a terminal or logs.
+type Result<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync + 'static>>;
+
+/// Representation of the rpm-ostree client-side state; this
+/// can be parsed directly from the output of `rpm-ostree status --json`.
+/// Currently not all fields are here, but that is a bug.
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+#[allow(unused)]
```

Comment:
Are the `allow(unused)` markers necessary even if we're returning the object from a public library function?